lib/timeline/component/item/Item.js

```diff
@@ -13,508 +17,528 @@ import '../css/item.css';
- * @return {Boolean} changed
- */
-Item.prototype.hide = function() {
-  return false;
-};
+  /**
+   * Hide the Item from the DOM (when visible)
+   * @return {Boolean} changed
+   */
+  hide() {
+    return false;
+  }
 
-/**
- * Repaint the item
- */
-Item.prototype.redraw = function() {
-  // should be implemented by the item
-};
+  /**
+   * Repaint the item
+   */
+  redraw() {
+    // should be implemented by the item
+  }
 
-/**
- * Reposition the Item horizontally
- */
-Item.prototype.repositionX = function() {
-  // should be implemented by the item
-};
+  /**
+   * Reposition the Item horizontally
+   */
+  repositionX() {
+    // should be implemented by the item
+  }
 
-/**
- * Reposition the Item vertically
- */
-Item.prototype.repositionY = function() {
-  // should be implemented by the item
-};
+  /**
+   * Reposition the Item vertically
+   */
+  repositionY() {
+    // should be implemented by the item
+  }
 
-/**
- * Repaint a drag area on the center of the item when the item is selected
- * @protected
- */
-Item.prototype._repaintDragCenter = function () {
-  if (this.selected && this.options.editable.updateTime && !this.dom.dragCenter) {
-    var me = this;
-    // create and show drag area
-    var dragCenter = document.createElement('div');
-    dragCenter.className = 'vis-drag-center';
-    dragCenter.dragCenterItem = this;
-    var hammer = new Hammer(dragCenter);
-
-    hammer.on('tap', function (event) {
-      me.parent.itemSet.body.emitter.emit('click',  {
-        event: event,
-        item: me.id
+  /**
+   * Repaint a drag area on the center of the item when the item is selected
+   * @protected
+   */
+  _repaintDragCenter() {
+    if (this.selected && this.options.editable.updateTime && !this.dom.dragCenter) {
+      const me = this;
+      // create and show drag area
+      const dragCenter = document.createElement('div');
+      dragCenter.className = 'vis-drag-center';
+      dragCenter.dragCenterItem = this;
+      this.hammerDragCenter = new Hammer(dragCenter);
+
+      this.hammerDragCenter.on('tap', event => {
+        me.parent.itemSet.body.emitter.emit('click',  {
+          event,
+          item: me.id
+        });
       });
-    });
-    hammer.on('doubletap', function (event) {
-      event.stopPropagation();
-      me.parent.itemSet._onUpdateItem(me);
-      me.parent.itemSet.body.emitter.emit('doubleClick', {
-        event: event,
-        item: me.id 
+      this.hammerDragCenter.on('doubletap', event => {
+        event.stopPropagation();
+        me.parent.itemSet._onUpdateItem(me);
+        me.parent.itemSet.body.emitter.emit('doubleClick', {
+          event,
+          item: me.id
+        });
       });
-    });
+      this.hammer.on('panstart', me.parent.itemSet._onDragStart.bind(me.parent.itemSet));
+      this.hammer.on('panmove',  me.parent.itemSet._onDrag.bind(me.parent.itemSet));
+      this.hammer.on('panend',   me.parent.itemSet._onDragEnd.bind(me.parent.itemSet));
```

Comment:
These should be `this.hammerDragCenter.*`
this.hammer does not exist.

Reply:
@mojoaxel This should fix the inability to edit items.

Reply:
Added